class-wpcom-compat-command.php

```diff
@@ -76,6 +76,103 @@ function import_protected_embeds( $args ) {
 			WP_CLI::line( sprintf( 'Failed to insert %d embeds', $errors ) );
 		}
 	}
+
+	/**
+	 * Parses existent protected embeds from WordPress.com
+	 *
+	 * This will replace all the protected-iframe shortcodes with the correct rendered embed
+	 *
+	 * # OPTIONS
+	 *
+	 * <file>
+	 *  : The CSV file with protected embeds
+	 *
+	 * @subcommand parse-protected-embeds
+	 */
+	function parse_protected_embeds( $args ) {
+		$file = $args[0];
+
+		if ( ! file_exists( $file ) ) {
+			WP_CLI::error( 'Specified file does not exist' );
+		}
+
+		$fd = fopen( $file, 'r' );
+		if ( ! $fd ) {
+			WP_CLI::error( sprintf( 'Could not open file: %s', $file ) );
+		}
+
+		$header = fgetcsv( $fd );
+		if ( ! is_array( $header ) || count( $header ) !== 6 ||
+			 ! in_array( 'id', $header, true ) ||
+			 ! in_array( 'embed_id', $header, true ) ||
+			 ! in_array( 'src', $header, true ) ||
+			 ! in_array( 'embed_group_id', $header, true ) ||
+			 ! in_array( 'html', $header, true ) ||
+			 ! in_array( 'time_added', $header, true ) ) {
+			WP_CLI::error( 'Invalid CSV, missing required fields' );
+		}
+
+		$protected_embeds = array();
+
+		// Main CSV loop
+		while ( $row = fgetcsv( $fd ) ) {
+			$data = array_combine( $header, $row );
+			if ( empty( $data ) || ! $data['id'] ) {
+				continue;
+			}
+
+			$protected_embeds[ $data['embed_id'] ] = $data;
+		}
+
+		// Remove all shortcodes and add the protected-iframes handler
+		global $shortcode_tags;
+		$shortcode_tags = array();
+
+		// Add custom protected-iframe handler
+		add_shortcode( 'protected-iframe', function ( $attrs ) use ( $protected_embeds ) {
+			return $protected_embeds[ $attrs['id'] ]['html'];
```

Comment:
Should add some error checking here. What if the shortcode doesn't have an ID or if the embed doesn't exist for some reason?

Reply:
I can't believe I forgot about that 🤦‍♂️. Will be fixed.